### PR TITLE
Modernization of the cmake in [Model]

### PR DESCRIPTION
**Pull request by knakul853**
I tried my hand at the "modernizing" model's Cmake. This is the first time I've worked with a build system (or rather a build system generator) so I might have made silly mistakes or misunderstood some things, so bear with me :).

I mostly just reconfigured the code to work at a target level instead of a directory level.

- follow the latest CMake version(3.16) syntax.
- add target-properties  bassed  syntax to CMake
- add multiline comments
- add more boost version 
- more CMake new feature [cmake](https://cliutils.gitlab.io/modern-cmake/)

### CODE REVIEW

**Comment by rcurtin** [comment]
Can you put descriptions with your PRs please?  For instance, why would we want to merge this change?  What is it even changing?  Why are the changes necessary?  Thanks. :+1:

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,10 +1,18 @@
-cmake_minimum_required(VERSION 2.6)
-project(models C CXX)
+cmake_minimum_required(VERSION 3.10)
+
+# set the project name and version
```
Comment (zoq) [inline CMakeLists.txt]: Looks like the idea here is to use some of the new CMake features, I think we can do a lot more similar to: https://github.com/mlpack/ensmallen/pull/152/files

Comment (knakul853) [reply CMakeLists.txt]: @zoq  yeah I mistakenly open the PR I was just playing with the codebase. Now i would complete this.

File: .travis.yml [hunk 2]
```diff
@@ -5,7 +5,7 @@ language: cpp
 before_install:
   - sudo apt-get update -qq
   - sudo apt-get install -qq libopenblas-dev liblapack-dev g++ libboost-all-dev
-  - curl https://ftp.fau.de/macports/distfiles/armadillo/armadillo-6.500.5.tar.gz | tar xvz && cd armadillo*
+  - curl http://sourceforge.net/projects/arma/files/armadillo-9.800.2.tar.xz | tar xvz && cd armadillo*
```
Comment (zoq) [inline .travis.yml]: You could use:

https://github.com/mlpack/ensmallen/blob/51614c216e43b235b97ff9059f571c8f02e3de37/.travis.yml#L16

as done in the ensmallen repo.

File: .travis.yml [hunk 3]
```diff
@@ -2,12 +2,20 @@ sudo: required
 dist: trusty
```
Comment (zoq) [inline .travis.yml]: Maybe if we run on Xenial it will help.

**Comment by knakul853** [comment]
Hey, I think the falling of build is not related to the CMake.what do you think ?

File: .travis.yml (hunk 3 above)
Comment (knakul853) [reply .travis.yml]: @zoq  tons of thanks : )

**Comment by knakul853** [comment]
@zoq @rcurtin  please add some comments : )

**Comment by knakul853** [comment]
I will be rewriting the CMake as @rcurtin  suggested.

**Pull request closed without merging**